robustel Blog (#1982)
FlowFuse/website: src/blog/2024/05/flowfuse-on-robustel-eg5120.md

---
title: Run FlowFuse on Robustel EG5120
subtitle: How to install FlowFuse Device Agent on Robustel EG5120
description: In this guide, we will discuss how to install FlowFuse Device agent on Robustel EG5120.

> date: 2024-05-07

Author: ```suggestion
date: 2024-05-08
```

---
title: Run FlowFuse on robustel EG5120
subtitle: How to install FlowFuse Device Agent on robustel EG5120
description: In this guide, we will discuss how to install FlowFuse Device agent on robustel EG5120.

> date: 2024-05-01

Author: ```suggestion
date: 2024-05-06
```

> date: 2024-05-06

Reviewer: ```suggestion
date: 2024-05-07
```

<!--more-->

This blog is designed for IoT enthusiasts and professionals who are looking to enhance their industrial IoT applications. The [robustel EG5120](https://www.robustel.com/product/eg5120-industrial-edge-computing-gateway/), equipped with Linux-based Debian 11 supporting a wide variety of programming languages including Node.js, offers robust connectivity options. When combined with FlowFuse, this gateway becomes even more powerful, enabling seamless device management and deployment. 

## robustel

The robustel EG5120 supports multiple connectivity options including Ethernet, Wi-Fi, and cellular networks, which are essential for flexible deployments in various industrial scenarios. Its built-in support for Bluetooth, cellular connectivity, RS232, RS485, and Modbus facilitates seamless integration with a wide array of IoT devices and services. This blog will guide you through using FlowFuse to effectively manage your Node-RED instance, enhancing both the security and scalability of your IoT applications.

## Migration

> If you're looking to migrate your Node-RED instance, follow the steps in [this migration guide](/docs/migration/node-red-tools/) to back up and transfer your configurations safely. By incorporating the FlowFuse device agent, we aim to replace the  Node-RED runtime, thus providing enhanced monitoring, automated backups, and improved management capabilities. This transition will make your deployments more secure and resilient.

Reviewer: As it's not installed by default, why migrate?

Author: Go point.  This was before I learned that I was working from a preconfigured device.  Some may still be using node-red as it does come preinstalled on NCD.io's devices.

Reviewer: But they don't google Robustel EG5120 Node-RED? Not sure why this comment was resolved but went unhandled. 

I'd be fine with an article outlining how NCD works that's more or less the same as this one, but this paragraph feels misplaced.

## Setting Up the Environment

> After spending a few weeks find out all the convoluted ways to do something, my CEO recommended testing out the raspbian deployment script for deploying FlowFuse to a Raspberry Pi.  Both Robustel and Raspberry pi run a debian base linux operating system.  Raspberry pi's run a customized and unofficial port of Debian called Raspbian, while Robustel leverages Debian 11.  I decided to give it a shot and the script works flawlessly.  

Reviewer 2: ```suggestion
After spending a few weeks finding out all the convoluted ways to do something, my CEO recommended testing the Raspbian deployment script for deploying FlowFuse to a Raspberry Pi.  Both Robustel and Raspberry pi run a debian base linux operating system.  Raspberry pi's run a customized and unofficial port of Debian called Raspbian, while Robustel leverages Debian 11.  I decided to give it a shot and the script works flawlessly.  
```

  
A simple script found (here)[/docs/hardware/raspbian/#installing-the-device-agent] can guide you on how to install FlowFuse on the EG5120.  

When finished you should see an out put stating **Started FlowFuse Device Agent.**

!["Started FlowFuse Device Agent"](./images/flowfuse-on-robustel-eg5120-installation-complete.png "Screenshot displaying completion of installation"){data-zoomable}

The script does the following:

1. Detect if Node.js is already installed, it will ensure it is at least v14. If less than v14 it will stop. If nothing is found it will install the Node.js 18 LTS release
2. Install the latest version of the FlowFuse Device Agent using npm.
3. Setup the FlowFuse Device Agent to run as a service and installs it to **/opt/flowfuse-device**.

> To run the agent a service continue following the documentation (here)[/docs/hardware/raspbian/#running-as-a-service].

Reviewer: Inverted brackets

> The last step is to link your EG5120 to FlowFuse.  These steps entail registering your device withing the FlowFuse platform.  The (guide here)[docs/device-agent/quickstart/] will help you register your EG5120 with FlowFuse.

Reviewer: Inverted brackets

> The last step is to link your EG5120 to FlowFuse.  These steps entail registering your device withing the FlowFuse platform.  The [guide here](docs/device-agent/quickstart/) will help you register your EG5120 with FlowFuse.

Author: ```suggestion
The last step is to link your EG5120 to FlowFuse.  These steps entail registering your device withing the FlowFuse platform.  The [guide here](/docs/device-agent/quickstart/) will help you register your EG5120 with FlowFuse.
```

## Setting Up the Environment

**Important Notice:** The device encounters an issue where the SuperUser and the default user are running different versions of Node.js. The SuperUser is running an outdated version incompatible with our needs. For the FlowFuse device agent, a minimum of NodeJS v14 is required, though v20 or later is recommended. For the latest recommendations, check this [documentation](/docs/device-agent/install/#prerequisites).

**1. Gaining Root Access:**
   - Start by opening your terminal and gain root access to adjust system configurations crucial for the following steps:
     ```bash
     sudo su
     ```
   - Verify the version of Node.js:
     ```bash
     node -v
     ```
   - If the version is below v14, proceed to the next step to upgrade. If it's v14 or higher, you might opt to skip to step 4.

**2. Updating Node.js:**

>    - To ensure optimal functionality, update or install Node.js v20.12 using the Node version manager:

Author: ```suggestion
   - To ensure optimal functionality, update or install Node.js LTS (Long Term Support) using the Node version manager:
```

     ```bash
     npm install n -g
     n lts
     ```

**3. Verifying Node.js Installation:**
   - Confirm the update by checking the Node.js version again:
     ```bash
     node -v
     ```
   - If the output confirms node version higher than v12, proceed to the next installation step.

**4. Installing FlowFuse Device Agent:**
   - Install the FlowFuse device agent globally to connect your robustel EG5120 to the FlowFuse platform efficiently:
     ```bash
     sudo npm install -g @flowfuse/device-agent
     ```

**4. Might need to turn off Node-RED service:**

> **Checking with the robustel team.**

Reviewer: When is this due?

---
title: Run FlowFuse on robustel EG5120
subtitle: How to install FlowFuse Device Agent on robustel EG5120
description: In this guide, we will discuss how to install FlowFuse Device agent on robustel EG5120.
date: 2024-05-01
authors: ["grey-dziuba"]
image: /blog/2024/05/images/flowfuse-on-rubustel-eg5120-node-red.png
tags:
    - posts
    - flowfuse
    - node.js
    - robustel
    - eg5120
---

The [robustel EG5120](https://www.robustel.com/product/eg5120-industrial-edge-computing-gateway/) is a versatile gateway that facilitates robust connectivity for industrial IoT applications. Integrating this powerful hardware with FlowFuse not only enhances its capabilities but also simplifies the management and deployment process. In this guide, we’ll walk through the steps to integrate the robustel EG5120 with FlowFuse, making your IoT solutions more efficient and easier to manage.

<!--more-->

> #### **Introduction**

Reviewer: I don't think we need to be saying this is the introduction.

```suggestion
```

This blog is designed for IoT enthusiasts and professionals who are looking to enhance their industrial IoT applications. The [robustel EG5120](https://www.robustel.com/product/eg5120-industrial-edge-computing-gateway/), equipped with Linux-based Debian 11 and preinstalled Node-RED, offers robust connectivity options. When combined with FlowFuse, this gateway becomes even more powerful, enabling seamless device management and deployment. 

> #### **Robustel and Node-RED**

Reviewer: `##` means H2, which is I think what we want and has better styling so no need to make the title bold

```suggestion
## Robustel and Node-RED
```

The robustel EG5120 supports multiple connectivity options including Ethernet, Wi-Fi, and cellular networks, which are essential for flexible deployments in various industrial scenarios. Its built-in support for MQTT and Modbus protocols facilitates seamless integration with a wide array of IoT devices and services. This blog will guide you through using FlowFuse to effectively manage your Node-RED instance, enhancing both the security and scalability of your IoT applications.

> #### **Migration**

Reviewer: ```suggestion
## Migration
```

If you're looking to migrate your Node-RED instance, follow the steps in [this migration guide](/docs/migration/node-red-tools/) to back up and transfer your configurations safely. By incorporating the FlowFuse device agent, we aim to replace the preinstalled Node-RED runtime, thus providing enhanced monitoring, automated backups, and improved management capabilities. This transition will make your deployments more secure and resilient.

> #### **Setting Up the Environment**

Reviewer: ```suggestion
## Setting Up the Environment
```

**Important Notice:** The device encounters an issue where the SuperUser and the default user are running different versions of Node.js. The SuperUser is running an outdated version incompatible with our needs. For the FlowFuse device agent, a minimum of NodeJS v14 is required, though v20 or later is recommended. For the latest recommendations, check this [documentation](/docs/device-agent/install/#prerequisites).

**1. Gaining Root Access:**
   - Start by opening your terminal and gain root access to adjust system configurations crucial for the following steps:
     ```bash
     sudo su
     ```
   - Verify the version of Node.js:
     ```bash
     node -v
     ```
   - If the version is below v14, proceed to the next step to upgrade. If it's v14 or higher, you might opt to skip to step 4.

**2. Updating Node.js:**

>    - To ensure optimal functionality, update or install Node.js v20.12 using the Node version manager:

Reviewer: Is there a way to install the latest stable version? That makes the post less temporal.

Author: The version of node.js that is v20.12 is LTS.  I can document that.

     ```bash
     npm install n -g
     n v20.12
     ```

**3. Verifying Node.js Installation:**
   - Confirm the update by checking the Node.js version again:
     ```bash
     node -v
     ```
   - If the output confirms v20.12, proceed to the next installation step.

> **4. Installing FlowFuse Device Agent:**

Reviewer: Did you test: https://flowfuse.com/docs/hardware/raspbian/#installing-the-device-agent? As it's both Debian based, it might be a nice one-line after the Node version has been upgraded.

Author: I haven't used that script yet.

Author: Yes.  This script works.  I am going this route.  My instructions before were not on a clean install like I thought.  This script worked like a charm and included it in the guide.

   - Install the FlowFuse device agent globally to connect your robustel EG5120 to the FlowFuse platform efficiently:
     ```bash
     sudo npm install -g @flowfuse/device-agent
     ```

**4. Might need to turn off Node-RED service:**

**Checking with the robustel team.**

**5. Device Registration and Initialization:**
   - After the installation of the device agent, it is now time to start the device agent:
     ```bash
     flowfuse-device-agent
     ```
   - Congratulation you now you now are utilizing FlowFuse to manage your Node-RED runtime.  If you would like this to run as a service please follow this [documentation](/blog/2023/05/device-agent-as-a-service/).

> #### **Visual Aids and Practical Application**

Reviewer: ```suggestion
## Visual Aids and Practical Application
```

Consider a scenario where a manufacturing facility needs to efficiently monitor and manage multiple industrial machines. By integrating the robustel EG5120 with FlowFuse, the facility can streamline data collection, enhance device management, and improve operational efficiencies through real-time analytics and remote management capabilities. Below are visual aids including diagrams and flowcharts that illustrate the integration process and clarify these complex concepts.

> #### **Troubleshooting Tips**

Reviewer: ```suggestion
## Troubleshooting Tips
```

During the integration process, you may encounter issues such as conflicting Node.js versions or connectivity problems. Here are some troubleshooting tips:
- Ensure all commands are run with appropriate permissions.
- Revisit the Node.js installation steps if version discrepancies occur.
- For connectivity issues, check network settings and device configuration.

> #### **Conclusion**

Reviewer: ```suggestion
## Conclusion
```
